### ISSUE TITLE
Build help with Allegro5 MinGW-64(TDM-Posix-SEH)

### ISSUE THREAD

**Issue by Axle-Ozz-i-sofT** [Immediate-Mode-UI/Nuklear#610]
Hi
I am attempting to get Nuklear to build using the Allegro5 back end. So far all of my attempts have failed and was wondering if anyone has a guide to do this.
Windows 10 x64
Target: x86_64-w64-mingw32
Thread model: posix
gcc version 9.2.0 (tdm64-1)

I am **NOT** using the Linux subsystem (MSYS2 etc.).
I typically set the paths for the GCC bin and dependent libraries in a batch file and call the Makefile with mingw32-make.
I am currently trying to build Nuklear from "allegro-x86_64-w64-mingw32-gcc-12.1.0-posix-seh-dynamic-5.2.8.0"
Nuklear-4.12.0\demo\allegro5\Makefile appears to be Linux shell orientated and the Windows_NT is commented out. The Windows_NT section as well as the Makefiles in other demos seams to target OpenGL and not Allegro, so I am a bit confused.
I tried altering the Variables to include the BIN ($BIN).exe and altered the mkdir to the windows command line syntax.
and
LIBS = -lallegro -lallegro_main -lallegro_image -lallegro_font \
	-lallegro_ttf -lallegro_primitives -lm

Always I get Error 2; The system cannot find the file specified.
```
rm -f bin/demo.exe
process_begin: CreateProcess(NULL, rm -f bin/demo.exe, ...) failed.
make (e=2): The system cannot find the file specified.
Makefile:21: recipe for target 'demo.exe' failed
mingw32-make: *** [demo.exe] Error 2
Press any key to continue . . .
```
I'm not sure what I am missing?

Also can Nuklear be built and used as a shared library (.so/.dll), or static only?

Thanks
Axle

P.S. I would actually be happy to just get any GCC/MinGW-w64 build working (DLL?) without Allegro just so I can orientate myself with Nuklear :)

**Comment by Axle-Ozz-i-sofT** [comment]
Update. I got the Nuklear-4.12.0\demo\allegro5\main.c to compile in my IDE.
Had to remove "-lallegro_main" as it isn't used. Otherwise the demo works fine with the Roboto-Regular.ttf and Allegro .dlls with the final executable.
>
I went back and altered the Makefile and got as far as "allegro5/allegro.h: No such file or directory" so I am thinking I have to rewrite this Makefile to hold the correct linker paths etc.

That being said I had intended to use Nuklear from the IDE anyway, so I can start learning the API as it is :)

Thank you
Axle

**Comment by Axle-Ozz-i-sofT** [comment]
Update:
Although it would be helpful for others to be able to use the supplied make files to test the library, it is not a big issue.
I jumped in and had a go at Nuklear and so far all good. I wasn't sure how to push my way through the docs and header API so I wrote a Visual Style Editor for Nuklear (Windows Allegro5 backed). I'll post a link when I have finished the Linux version.
I'll close this issue :)
Regards
Axle
